src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -406,6 +416,9 @@ private boolean canDoExactSearch(final int filterIdsCount) {
             filterIdsCount,
             KNNSettings.getFilteredExactSearchThreshold(knnQuery.getIndexName())
         );
+        if (knnQuery.getRadius() >= 0) {
```

Comment:
What are we doing here?

Reply:
for native engines radial search we juts do ANNSearch, so if the radial parameter is defined then here return false for ExactSearch

Reply:
have we added a test for this? and are we going to raise another PR where the radial search will be added for exact search too?

Reply:
@navneet1v Will add test. Exact search isn’t in scope of this release.

Reply:
It should be in scope otherwise the relevancy for filtered radial search will go down. I would recommend adding that. If you want to handle it in another PR that is also fine.

Reply:
Got it, we can discuss to add it in the scope. cc @vamshin